xbmc/windowing/gbm/GBMUtils.h

```diff
@@ -158,7 +158,7 @@ class CGBMUtils
    *
    * @return CGBMDevice* A pointer to the CGBMDevice object
    */
-  CGBMUtils::CGBMDevice* GetDevice() const { return m_device.get(); }
+  CGBMUtils::CGBMDevice& GetDevice() const { return *m_device; }
```

Comment:
This looks problematic because m_device is not allocated in the constructor. If `CreateDevice()` is never called then this can be null.

Reply:
I don't think there is much difference here currently because the returned pointers aren't checked anyways.

Reply:
if m_device must exist then this is fine, but this is usually guaranteed be allocating m_device in the constructor. Can we allocate early and then call an Initialize() function, so that GetDevice() would return an uninitialized device instead of choking on a null ptr dereference?